src/cms/src/helpers/validator.js

```diff
@@ -2,6 +2,18 @@ const utils = require('@strapi/utils');
 const { ApplicationError } = utils.errors;
 
 module.exports = {
+  // saving without a protectedArea relation is not allowed
+  protectedAreaConnectValidator: function (protectedArea) {
+    if (protectedArea.connect.length === 0) {
+      throw new ApplicationError('Please add protectedArea relation.');
+    }
+  },
+  // removing a protectedArea relation is not allowed
+  protectedAreaDisconnectValidator: function (protectedArea) {
+    if (protectedArea.disconnect.length > 0) {
```

Comment:
This doesn't work right when you change to a different protectedArea.  Maybe try something like this instead?
```
    if (protectedArea.disconnect.length > 0 && protectedArea.connect.length === 0) {
```

Reply:
That works, thanks!